### PR TITLE
Allow execution of deploy command without docker

### PR DESCRIPTION
**Pull request by victorpoluceno**
To make easier to test and better tutorials we are making possible to run the deploy command locally. Also, this is getting rid of the cf-step-helm dependency. I also took the time to make some more parameters non requirement.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,18 +12,18 @@ COPY prune/* ./
 RUN go build -o ./prune .
 RUN chmod +x prune
 
-# Helm #
-FROM codefresh/cfstep-helm:3.2.4
+FROM alpine:3.13.5
```
Comment (victorpoluceno) [inline Dockerfile:15]: We don't need cf-step-helm we can just use alpine.

File: Dockerfile [hunk 2]
```diff
@@ -36,6 +36,12 @@ RUN curl -LO --silent https://storage.googleapis.com/kubernetes-release/release/
     && mv ./kubectl /usr/local/bin/kubectl \
     && chmod +x /usr/local/bin/kubectl
 
+# Helm #
```
Comment (victorpoluceno) [inline Dockerfile:39]: Install helm so everything keeps working.

File: deploy/deploy [hunk 3]
```diff
@@ -9,7 +9,6 @@ fi
 
 # Helm Chart variables
 chart_version=${KR_CHART_VERSION:-0.5}
-helm_version=${KR_HELM_VERSION:-3.2.4}
```
Comment (victorpoluceno) [inline deploy/deploy:12]: This was not in use.

File: deploy/deploy [hunk 4]
```diff
@@ -25,16 +24,16 @@ repo_owner=$KR_REPO_OWNER
 
 # Kube Review variables
 prefix=${KR_PREFIX:-re}
+helm_repo=${KR_HELM_REPO_URL:-"cm://h.cfcr.io/findhotel/default"}
```
Comment (victorpoluceno) [inline deploy/deploy:27]: Setting default value so it's one less variable.

File: deploy/deploy [hunk 5]
```diff
@@ -25,16 +24,16 @@ repo_owner=$KR_REPO_OWNER
...
 name="$prefix-${KR_ID_OVERRIDE:-$KR_ID}"
-short_name=$(echo -n $name | cut -c1-15)
+short_name=$(echo "$name" | cut -c1-15)
 hash=$(echo "$name" | rhash -p "%c" -)
 namespace=$short_name-$hash
 is_ephemeral=${KR_IS_EPHEMERAL:-true}
 host=$namespace.$KR_DOMAIN
 url=https://$host
 kube_context=$KR_KUBE_CONTEXT
-kube_post_install_msg=$KR_POST_INSTALL_MSG
```
Comment (victorpoluceno) [inline deploy/deploy:37]: Not used anymore.

File: deploy/deploy [hunk 6]
```diff
@@ -44,12 +43,15 @@ config_context() {
   scoped_kubeconfig_file=$kube_config_file-$namespace
   cp "$kube_config_file" "$scoped_kubeconfig_file"
   export KUBECONFIG=${scoped_kubeconfig_file}
-  kubectl config use-context "$kube_context"
+
+  if [ "$kube_context" != "" ]; then
```
Comment (victorpoluceno) [inline deploy/deploy:47]: Make kube context non mandatory, as a context might already be selected.

File: deploy/deploy [hunk 7]
```diff
@@ -44,12 +43,15 @@ config_context() {
...
+    kubectl config use-context "$kube_context"
+  fi
 }
 
 namespace_apply() {
   # Variables need to be export in order to be replaced by envsubst
-  updated_at=$(date --utc +%s)
```
Comment (victorpoluceno) [inline deploy/deploy:52]: This flag is not compatible with bash.

File: deploy/deploy [hunk 8]
```diff
@@ -107,47 +109,38 @@ uninstall_chart() {
   # There is a problem in Helm v3 when it is necessary to update the helm graph
   # sometimes an UPGRADE_FAILED occurs, to fix it, we first of all need to remove the old version
   echo "Uninstalling old version of the helm chart: $namespace"
-  helm uninstall -n "$namespace" "$namespace"
+  (set -x; helm uninstall -n "$namespace" "$namespace")
```
Comment (victorpoluceno) [inline deploy/deploy:112]: `set -x` to show the command in the sdtout.

File: deploy/deploy [hunk 9]
```diff
@@ -107,47 +109,38 @@ uninstall_chart() {
...
 }
 
 install_chart() {
+  (set -x; helm plugin install https://github.com/chartmuseum/helm-push.git || true)
+  (set -x; helm repo add remote "$helm_repo")
+
   if [ "$KR_VALUES_FILE" != "" ]; then
     echo "Using values file from user: $KR_VALUES_FILE"
-    export CUSTOMFILE_0="$KR_VALUES_FILE"
+    opts+=(--values "$KR_VALUES_FILE")
```
Comment (victorpoluceno) [inline deploy/deploy:121]: This is how we handle optional options.

File: deploy/deploy [hunk 10]
```diff
@@ -107,47 +109,38 @@ uninstall_chart() {
...
-  CUSTOM_image_repository="$KR_IMAGE_URL" \
-  CUSTOM_envFrom_secretRef_name="$namespace" \
-  CUSTOM_"ingress_hosts[0]_host=$host" \
-  CUSTOM_"ingress_hosts[0]_paths[0]=/" \
-  CUSTOM_"ingress_tls[0]_hosts[0]=$host" \
-  SKIP_CF_STABLE_HELM_REPO=true \
-  WAIT=true \
-  CMD_PS='--cleanup-on-fail' \
-  /opt/bin/release_chart
+  (set -x; helm upgrade "$namespace" \
```
Comment (victorpoluceno) [inline deploy/deploy:124]: This is where we don't rely on cf-step-helm script anymore.

File: deploy/deploy [hunk 11]
```diff
@@ -166,7 +159,9 @@ export_variables() {
 }
 
 print_postinstall_message() {
-  printf "%s\n" "$KR_MESSAGE" "$KR_LOG_URL$namespace"
+  if [ "$KR_MESSAGE" != "" ]; then
```
Comment (victorpoluceno) [inline deploy/deploy:162]: Avoids printing of namespace when there is no message to print.

File: Dockerfile [hunk 12]
```diff
@@ -36,6 +36,12 @@ RUN curl -LO --silent https://storage.googleapis.com/kubernetes-release/release/
...
+RUN curl -L --silent https://get.helm.sh/helm-${HELM_VERSION}-linux-386.tar.gz -o helm.tar.gz \
+    && tar -zxf helm.tar.gz \
+    && mv linux-386/helm /usr/local/bin/helm \
+    && chmod +x /usr/local/bin/helm
+
 WORKDIR /
 COPY deploy/* ./
 COPY --from=base /prune ./
```
Comment (rlinstorres) [inline Dockerfile:47]: Do we still need to use it?
```
# We need to do this because cfstep-helm has an entrypoint
ENTRYPOINT []
`` `

**Review by rlinstorres — COMMENTED** [review]
I left small comments

**Review by gtonello — APPROVED** [review]
Nice work! LGTM

File: deploy/deploy (hunk 4 above)
Comment (victorpoluceno) [reply deploy/deploy]: yeah, but then users calling the /deploy script directly will have to pass the argument. So, I would like to make it optional. It's not an issue to have the codefresh repo as it's not specific to cf, if we move away from cf we just need to replace the url with one of the new repo. So, I think it does make sense to have this default value here.

**Pull request merged**